File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int numMinus = 0;
        for(int i : nums) {
            if( i == 0)
                return 0;
            else if(i < 0) numMinus++;
        }
        return (numMinus%2 == 0)?1:-1;
        
    }
}

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int> &nums)
  {
    int result = 1;
    for (int i = 0; i < nums.size(); i++)
    {
      result *= ((nums[i] > 0) ? 1 : (nums[i] == 0) ? 0
                                                    : -1);
      if (result == 0)
        return 0;
    }
    return result;
  }
};